lib/File/Temp.pm

```diff
@@ -164,7 +164,13 @@ eval { require Carp::Heavy; };
 require Symbol if $] < 5.006;
 
 ### For the OO interface
-use parent 0.221 qw/ IO::Handle IO::Seekable /;
+#use parent 0.221 qw/ IO::Handle IO::Seekable /; ## Win32 Perl 5.8 is missing "parent"
```

Comment:
The proper way to fix this is to ensure that `parent` is installed as a prerequisite.

Reply:
I think there is a circular dependency on 5.8 here (via Test::More, and possibly other routes). This high up the river, it's usually better to `use base`.

Reply:
So the issue is that CORE modules are/have been generally tested without any dependency installation (which I think is reasonable practice). As I said in the initial post and the comment, MSWin32 perl v5.8 doesn't have "parent" in CORE, so using "parent" introduces a non-CORE dependency.

I understand that "parent" has become the standard idiom which is one of the reasons I left the original code as a comment and added the "missing 'parent'" comment.

I could handle this in a couple of different ways:

1. Replace "parent" with an equivalent which works with all perl versions (as I did here)
2. Add a specific extra dependency that is installed just for the failing perl versions, but doesn't actually test in a CORE-like fashion.
3. Skip testing for MSWin32 perl v5.8 entirely.

I chose #1.

I could change to option #2 by altering the AppVeyor config or by adding an .appveyor_init file (I did this for "version" because I "parent" was used in multiple places and I wasn't completely sure what the equivalent code would be [see <https://github.com/Perl/version.pm/pull/3/files#diff-380ca652b9d9d0db969f3d51e13e26ce>].

I think option #3 is the least desirable, especially given the relatively simple repair shown in the PR.

The current PR works and passes all tests for all versions (both MSWin32 and *nix; see reports in the initial post).

Reply:
`parent` is already in the list of prerequisites for this module. Why is your 5.8 install not detecting that and resolving it like any other prereq?  https://metacpan.org/source/ETHER/File-Temp-0.2309/META.json#L69

Reply:
I just explained in the previous post.
As a CORE module, dependencies are not being tested or installed.
If you would prefer option #2 above instead of option #1, I can change it as I said.

Reply:
> As a CORE module, dependencies are not being tested or installed.

I'm still not understanding what piece of software is installing a module without installing its dependencies. This is a serious problem that should be addressed, wherever it is.

Reply:
MSWin32 perl v5.8 doesn't come with "parent" in CORE... it's just not in the distribution.

File::Temp is a CORE module.

When testing CORE modules on AppVeyor, as per other CORE modules, no testing for or installation of prerequisite dependencies is done (using `DIST_SUPPRESS_DEPS=1` in the *.appveyor_init.BAT* file) the CORE module (File::Temp here) is expected to work without installing anything else beyond the basic perl distribution.

But, File::Temp requires "parent" which is not in MSWin32 perl v5.8 and fails testing due to lack of that dependency.

There is no software failure. The testing is conducted in a "don't check for or install any dependencies prior to testing" mode. I can force "parent" to be installed even in that mode (see option #2 above...; in the manner shown for the "version" module, also noted above).

Reply:
> When testing CORE modules on AppVeyor, as per other CORE modules, no testing for or installation of prerequisite dependencies is done

That's a bug in AppVeyor then.  File::Temp is dual-life and as such can have other dependencies that are also dual-lifed (like parent).  It's absolutely legal for an upgrade of a dual-life module to pull in other dependencies that are also upgrades (not in core at that version, or even not in core at all for that particular version of perl).

Reply:
Nope, not a bug in "AppVeyor". If there's any "bug", it's in the Strawberry Perl v5.8 distribution which doesn't come with "parent". But, it's the only distribution that's available for testing.

File::Temp is not being "installed", only tested, and it is, as noted, *purposefully* being tested without adding any dependencies to the basic perl installation (or even checking for any dependencies). It's being tested in the bare CORE environment of MSWin32 v5.8, for the reasons I've already spoken about.

Given that this repository is in a `dzil`-based form and `dzil` won't even run in v5.8, and `cpanm`/`cpan` can't parse the respository, as is, for dependencies, I'm not even sure how to automatically find dependencies anyway.

But, if you don't like the reasons given, and if you want "parent" to be automatically brought in for the testing, I can change it in the configuration by, as I've repeatedly said, adding an additional step to pull in that dependency.

I've expressed my opinion, but it's your module. And you obviously vehemently disagree with the fix that I've used. It is, however, completely changeable. But I must not be explaining my reasons or the alternative options well enough. So, pick #2 or #3 above, and I'll change to that method. Or, I can just drop the commit.

Or, if, as it seems, you seem to find this more bothersome than valuable, then refuse and close the PR. I spent the time creating a this AppVeyor config/helpers to be actually helpful, not pick a fight.

Reply:
> File::Temp is not being "installed", only tested, and it is, as noted, _purposefully_ being tested without adding any dependencies to the basic perl installation (or even checking for any dependencies). It's being tested in the bare CORE environment of MSWin32 v5.8, for the reasons I've already spoken about.

That decision is incorrect. Tests cannot be run without satisfying prerequisites (see https://metacpan.org/pod/CPAN::Meta::Spec#Phases), and there *are* prerequisites. `parent` isn't the only one that isn't satisfied by a bare 5.8 install, either - there is also  Exporter 5.57 (5.8.0 only has 1.01), and File::Path 2.06 (5.8.0 only has 1.05).  If AppVeyor on Strawberry Perl can't properly test against 5.8, it should skip that version, rather than trying to force an invalid installation configuration. But prerequisites absolutely *must* be satisfied first, and it wouldn't be a valid test to skip that step.

> Given that this repository is in a `dzil`-based form and `dzil` won't even run in v5.8, and `cpanm`/`cpan` can't parse the respository, as is, for dependencies, I'm not even sure how to automatically find dependencies anyway.

Ah, that's a different problem, and one that can be solved (e.g. by putting a standalone Makefile.PL or cpanfile in the repository). It hasn't been needed so far because this is a pure-perl distribution, and the travis configuration uses a more recent perl version to build the distribution first before switching to the intended perl version for actual testing.

Reply:
I committed a cpanfile into master - you should be able to get at the prereqs on 5.8 now.